### ISSUE TITLE
Single query for aggregate?

### ISSUE THREAD

**Issue by fjsj** [douwevandermeij/admin-totals#3]
### EDIT: This is wrong! See below my last comment.

Hi, thanks for this library! It worked well for me.

One suggestion: have you considered making a single query for aggregates instead of one query per aggregate?

I've adapted the code to do this:
```python
from django.contrib.admin.views.main import ChangeList

class ChangeListTotals(ChangeList):

    def get_results(self, request):
        super().get_results(request)
        if hasattr(self.model_admin, 'list_totals'):
            list_totals = dict(self.model_admin.list_totals)

            aggregation_functions = {}
            for field in list_totals:
                aggregation_functions[field] = list_totals[field](field)

            aggregation_results = self.result_list.aggregate(**aggregation_functions)

            self.aggregations = []
            for field in self.list_display:
                if field in aggregation_results:
                    self.aggregations.append(aggregation_results[field])
                else:
                    self.aggregations.append('')
```

**Comment by douwevandermeij** [comment]
Hi there, thanks for pointing out. Sounds like a good feature.
You mind to create a pull request for this?

**Comment by fjsj** [comment]
I'll do it. Thanks.

**Comment by fjsj** [comment]
This idea is wrong. Thankfully I did TDD here!
TL;DR: Avg + Count of m2m breaks due to joins.

[Tests](https://github.com/vintasoftware/admin-totals/blob/1d4b9ad728abdace22e1cdac184a525e3d2a6eca/tests/admin_changelist/tests.py) break if we change the code to do a single query:
```
FAIL: test_aggregations (tests.admin_changelist.tests.ChangeListTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/fjsj/workspace/admin-totals/tests/admin_changelist/tests.py", line 104, in test_aggregations
    self._test_aggregations_django2()
  File "/Users/fjsj/workspace/admin-totals/tests/admin_changelist/tests.py", line 96, in _test_aggregations_django2
    5
AssertionError: Lists differ: ['', '', 5.6, 5] != ['', '', 5.5, 5]

First differing element 2:
5.6
5.5

- ['', '', 5.6, 5]
?            ^

+ ['', '', 5.5, 5]
?            ^
```

If you run the raw query, you see why:
```
>>> with connection.cursor() as cursor:
...     cursor.execute('''
...         SELECT AVG("admin_changelist_band"."nr_of_members") AS "nr_of_members", COALESCE(COUNT("admin_changelist_band_genres"."genre_id"), 0) AS "genres" FROM "admin_changelist_band" LEFT OUTER JOIN "admin_changelist_band_genres" ON ("admin_changelist_band"."id" = "admin_changelist_band_genres"."band_id")
...     ''')
...     print(cursor.fetchall())
... 
[(Decimal('5.6000000000000000'), 5)]
>>> 
>>> with connection.cursor() as cursor:
...     cursor.execute('''
...         SELECT "nr_of_members", "admin_changelist_band_genres"."genre_id" FROM "admin_changelist_band" LEFT OUTER JOIN "admin_changelist_band_genres" ON ("admin_changelist_band"."id" = "admin_changelist_band_genres"."band_id")
...     ''')
...     print(cursor.fetchall())
... 
[(5, 1), (5, 2), (6, 3), (6, 4), (6, 5)]
```
